src/Socket_p.h

```diff
@@ -573,7 +573,7 @@ namespace Arcus
 
         {
             std::lock_guard<std::mutex> lk(receiveQueueMutexBlock);
-            message_received_condition_variable = true;
+            message_received_condition_variable.notify_all();
```

Comment:
Same as earlier: Is there a reason this is 'notify_all' instead of 'notify_one'?